tools/distrib/fix_build_deps.py

```diff
@@ -0,0 +1,410 @@
+    'openssl/ssl.h':
+        'libssl',
+    'openssl/tls1.h':
+        'libssl',
+    'openssl/x509.h':
+        'libcrypto',
+    'openssl/x509v3.h':
+        'libcrypto',
+    're2/re2.h':
+        're2',
+    'upb/def.h':
+        'upb_lib',
+    'upb/json_encode.h':
+        'upb_json_lib',
+    'upb/text_encode.h':
+        'upb_textformat_lib',
+    'upb/def.hpp':
+        'upb_reflection',
+    'upb/upb.h':
+        'upb_lib',
+    'upb/upb.hpp':
+        'upb_lib',
+    'xxhash.h':
+        'xxhash',
+    'zlib.h':
+        'madler_zlib',
+}
+
+INTERNAL_DEPS = {
+    'google/rpc/status.upb.h':
+        'google_rpc_status_upb',
+    'google/protobuf/any.upb.h':
+        'protobuf_any_upb',
+    'google/protobuf/duration.upb.h':
+        'protobuf_duration_upb',
+    'google/protobuf/struct.upb.h':
+        'protobuf_struct_upb',
+    'google/protobuf/timestamp.upb.h':
+        'protobuf_timestamp_upb',
+    'google/protobuf/wrappers.upb.h':
+        'protobuf_wrappers_upb',
+    'src/proto/grpc/channelz/channelz.grpc.pb.h':
+        '//src/proto/grpc/channelz:channelz_proto',
+    'src/proto/grpc/core/stats.pb.h':
+        '//src/proto/grpc/core:stats_proto',
+    'src/proto/grpc/health/v1/health.upb.h':
+        'grpc_health_upb',
+    'src/proto/grpc/lb/v1/load_reporter.grpc.pb.h':
+        '//src/proto/grpc/lb/v1:load_reporter_proto',
+    'src/proto/grpc/lb/v1/load_balancer.upb.h':
+        'grpc_lb_upb',
+    'src/proto/grpc/reflection/v1alpha/reflection.grpc.pb.h':
+        '//src/proto/grpc/reflection/v1alpha:reflection_proto',
+    'src/proto/grpc/gcp/transport_security_common.upb.h':
+        'alts_upb',
+    'src/proto/grpc/gcp/altscontext.upb.h':
+        'alts_upb',
+    'src/proto/grpc/lookup/v1/rls.upb.h':
+        'rls_upb',
+    'src/proto/grpc/lookup/v1/rls_config.upb.h':
+        'rls_config_upb',
+    'src/proto/grpc/lookup/v1/rls_config.upbdefs.h':
+        'rls_config_upbdefs',
+    'src/proto/grpc/testing/xds/v3/csds.grpc.pb.h':
+        '//src/proto/grpc/testing/xds/v3:csds_proto',
+    'xds/data/orca/v3/orca_load_report.upb.h':
+        'xds_orca_upb',
+    'xds/service/orca/v3/orca.upb.h':
+        'xds_orca_service_upb',
+    'xds/type/v3/typed_struct.upb.h':
+        'xds_type_upb',
+}
+
+SKIP_DEPS = {'google/api/expr/v1alpha1/syntax.upb.h'}
+
+
+class FakeSelects:
+
+    def config_setting_group(self, **kwargs):
+        pass
+
+
+def grpc_cc_library(name,
+                    hdrs=[],
+                    public_hdrs=[],
+                    srcs=[],
+                    select_deps=None,
+                    tags=[],
+                    **kwargs):
+    if select_deps or 'nofixdeps' in tags or 'grpc-autodeps' not in tags:
+        no_update.add(name)
+    score = len(public_hdrs + hdrs)
+    if 'avoid_dep' in tags:
+        score += 1000000
+    if 'nofixdeps' in tags:
+        score += 1000
+    for hdr in hdrs + public_hdrs:
+        vendors.setdefault(hdr, []).append(Vendor(name, score))
```

Comment:
Nit: You could use [`collections.defaultdict(list)`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile) instead.

Reply:
done